tools/integration_tests/util/client/storage_client.go

```diff
@@ -222,3 +223,59 @@ func StatObject(ctx context.Context, client *storage.Client, object string) (*st
 	}
 	return attrs, nil
 }
+
+func UploadGcsObject(localPath, bucketName, objectName string, uploadGzipEncoded bool) error {
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create storage client: %v", err)
+	}
+	defer client.Close()
+
+	// Open the local file for reading
+	f, err := os.Open(localPath)
```

Comment:
Use https://github.com/GoogleCloudPlatform/gcsfuse/blob/8aea5961f83c74fdbf5d4326d585c96d10951dad/tools/integration_tests/util/operations/file_operations.go#L327 here (you'd need to also make that method public in the file_operations.go for reuse here).